rcl/src/rcl/node.c

```diff
@@ -179,7 +211,47 @@ rcl_node_init(
   }
   // actual domain id
   node->impl->actual_domain_id = domain_id;
-  node->impl->rmw_node_handle = rmw_create_node(name, local_namespace_, domain_id);
+
+  const char * ros_enable_security = NULL;
+  const char * ros_enforce_security = NULL;
+
+  if (rcutils_get_env(sros_enable_security_var_name, &ros_enable_security)) {
```

Comment:
Not that it matters, but it is only not thread-safe on Windows. You could make a thread safe version by making this global thread-local:

https://github.com/ros2/rcutils/blob/master/src/get_env.c#L27